docs/creating-a-runtime-check.md

```diff
@@ -0,0 +1,181 @@
+# Creating a Runtime Check
+
+Runtime checks are used to perform tests by executing the plugins code.
+
+The Plugin Checker uses a number of interfaces to build out runtime checks.
+
+One interface is the `Runtime_Check` interface. This interface contains no methods, but is used to help identify a runtime check from static checks.
+
+In addtion, there is also the `Preparation` interface, which is used to define checks that have preparations. This interface defines a `prepare()` method which is used to run the logic required to prepare the environment before the check before is run.
+
+Both these interfaces are implemented in the `Abstract_Runtime_Check` class, which developers can use when building out their own runtime checks. This class defines both the `prepare()` and `run()` methods which are required to be implemented for every runtime check.
+
+Below is the basic scaffold when creating a custom runtime check.
+
+```php
+use WordPress\Plugin_Check\Checker\Checks\Abstract_Runtime_Check;
+
+class Custom_Check extends Abstract_Runtime_Check {
+  public function prepare() {
+    // Handle the Checks preparations and return a cleanup function.
+  }
+
+  public function run( Check_Results $results ) {
+    // Handle running the check and adding warnings or errors to the results.
+  }
+}
+```
+
+## Preparations
+
+Preparations are used to prepare the environment before running the check. Preparations can include any logic from activating specific themes or plugins, or creating test content required to perform the check.
+
+Preparations will use the `Preparation` interface which defines a single `prepare()` method. This method is required to be implemented in runtime checks and handles any logic needed to prepare the environent.
+
+All preparations return a clean up function that reverts the changes made by the method.
+
+When creating preparations there are 3 approachs that can be used depending on circumstance.
```

Comment:
You explain a lot of this already in the technical overview. At a minimum, we should align the two sections to basically have the same content, or alternatively we shorten the one in the technical overview to just the most important thing, like 1-2 sentences.